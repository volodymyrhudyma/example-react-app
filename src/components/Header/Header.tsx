import React, { Component, ChangeEvent } from 'react';
import { Wrapper, HeaderInner } from './styled/Header.styled';
import Input from 'src/ui-kit/Input';
import Avatar from 'src/ui-kit/Avatar';
import icons from 'src/constants/icons';
import Container from 'src/ui-kit/Container';
import { WithTranslation } from 'react-i18next';

interface IHeaderState {
  query: string;
}

class Header extends Component<WithTranslation, IHeaderState> {
  constructor(props: WithTranslation) {
    super(props);
    this.state = {
      query: '',
    };
  }
  onInput = (e: ChangeEvent<HTMLInputElement>) => {
    const { value } = e.target;
    this.setState({
      query: value,
    });
  };
  render() {
    const { t } = this.props;
    const { query } = this.state;
    return (
      <Wrapper>
        <Container>
          <HeaderInner>
            <Input
              value={query}
              onChange={this.onInput}
              placeholder={t('search.placeholder')}
            />
            <Avatar src={icons.headerAvatar} />
          </HeaderInner>
        </Container>
      </Wrapper>
    );
  }
}

export default Header;
