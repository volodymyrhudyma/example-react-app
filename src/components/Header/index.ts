import Component from './Header';
import { withTranslation } from 'react-i18next';

export default withTranslation()(Component);
