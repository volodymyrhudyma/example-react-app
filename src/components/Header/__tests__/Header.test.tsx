import { shallow } from 'enzyme';
import React from 'react';
import Header from '../Header';
import i18n from 'i18next';

describe('Header with Enzyme and Jest', () => {
  it('should render correctly', () => {
    const component = shallow(
      <Header t={jest.fn()} i18n={i18n} tReady={false} />
    );
    expect(component).toMatchSnapshot();
  });
});
