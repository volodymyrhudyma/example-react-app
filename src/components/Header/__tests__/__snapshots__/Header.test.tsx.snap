// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header with Enzyme and Jest should render correctly 1`] = `
<styled.div>
  <Container>
    <styled.div>
      <Input
        onChange={[Function]}
        type="text"
        value=""
      />
      <Avatar
        height={60}
        src="/icons/avatar-header.png"
        width={60}
      />
    </styled.div>
  </Container>
</styled.div>
`;
